presto-memory/src/main/java/io/prestosql/plugin/memory/MemoryMetadata.java

```diff
@@ -358,4 +359,17 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
     {
         return ImmutableList.copyOf(tables.get(tableId).getDataFragments().values());
     }
+
+    @Override
+    public Optional<ConnectorTableHandle> applyLimit(ConnectorTableHandle handle, long limit)
+    {
+        MemoryTableHandle table = (MemoryTableHandle) handle;
+
+        if (table.getLimit().isPresent() && table.getLimit().getAsLong() < limit) {
+            // can't apply the limit if we already have a lower limit
```

Comment:
I don't think that is right.  If we are only going to produce 100 rows, and there is a limit of 200, we are still good.